src/WasmExecutor.cpp

```diff
@@ -340,208 +420,82 @@ auto dynamic_type_dispatch(const halide_type_t &type, Args &&... args) -> declty
         using ReturnType = decltype(std::declval<Functor<uint8_t>>()(std::forward<Args>(args)...));
         return ReturnType();
     }
-#undef HANDLE_CASE
-}
-
-// ------------------------------
-
-template<typename T>
-struct ExtractAndStoreScalar {
-    void operator()(const Local<Context> &context, const Local<Value> &val, void *slot) {
-        *(T *)slot = (T)val->NumberValue(context).ToChecked();
-    }
-};
-
-template<>
-inline void ExtractAndStoreScalar<float16_t>::operator()(const Local<Context> &context, const Local<Value> &val, void *slot) {
-    float16_t f((double)val->NumberValue(context).ToChecked());
-    *(uint16_t *)slot = f.to_bits();
-}
-
-template<>
-inline void ExtractAndStoreScalar<bfloat16_t>::operator()(const Local<Context> &context, const Local<Value> &val, void *slot) {
-    bfloat16_t b((double)val->NumberValue(context).ToChecked());
-    *(uint16_t *)slot = b.to_bits();
-}
-
-template<>
-inline void ExtractAndStoreScalar<void *>::operator()(const Local<Context> &context, const Local<Value> &val, void *slot) {
-    internal_error << "TODO: 64-bit slots aren't yet supported";
-}
 
-template<>
-inline void ExtractAndStoreScalar<uint64_t>::operator()(const Local<Context> &context, const Local<Value> &val, void *slot) {
-    internal_error << "TODO: 64-bit slots aren't yet supported";
+#undef HANDLE_CASE
 }
+// clang-format on
 
-template<>
-inline void ExtractAndStoreScalar<int64_t>::operator()(const Local<Context> &context, const Local<Value> &val, void *slot) {
-    internal_error << "TODO: 64-bit slots aren't yet supported";
+std::string to_string(const wabt::MemoryStream &m) {
+    // TODO: ugh
```

Comment:
Please elaborate. Are you talking about the weird requirement to const cast?

Reply:
Yeah, that comment doesn't belong there. Removing.